retrieval_service/models/models.py

```diff
@@ -38,9 +38,43 @@ class Amenity(BaseModel):
     terminal: str
     category: str
     hour: str
+    sunday_start_hour: Optional[datetime.time]
+    sunday_end_hour: Optional[datetime.time]
+    monday_start_hour: Optional[datetime.time]
+    monday_end_hour: Optional[datetime.time]
+    tuesday_start_hour: Optional[datetime.time]
+    tuesday_end_hour: Optional[datetime.time]
+    wednesday_start_hour: Optional[datetime.time]
+    wednesday_end_hour: Optional[datetime.time]
+    thursday_start_hour: Optional[datetime.time]
+    thursday_end_hour: Optional[datetime.time]
+    friday_start_hour: Optional[datetime.time]
+    friday_end_hour: Optional[datetime.time]
+    saturday_start_hour: Optional[datetime.time]
+    saturday_end_hour: Optional[datetime.time]
     content: Optional[str] = None
     embedding: Optional[list[float]] = None
 
+    @field_validator(
+        "sunday_start_hour",
+        "sunday_end_hour",
+        "monday_start_hour",
+        "monday_end_hour",
+        "tuesday_start_hour",
+        "tuesday_end_hour",
+        "wednesday_start_hour",
+        "wednesday_end_hour",
+        "thursday_start_hour",
+        "thursday_end_hour",
+        "friday_start_hour",
+        "friday_end_hour",
+        "saturday_start_hour",
+        "saturday_end_hour",
+        mode="before",
+    )
+    def replace_none(cls, v):
+        return v or None
```

Comment:
Is this still neccesary with an optional field?

Reply:
Yes, within the csv dataset, these `None` value are empty. For example: 
Sunday 9-10pm, 09:00:00, 22:00:00,,,,, (those `None` value is converted as just empty).
Hence, when running validation, if without replacing them with actual `None`, it will throw an error saying that the type is different.

Reply:
I think that's okay, but we probably need to update the retrieval service to say "closed" as opposed to None in the string it returns.

Reply:
added a bug for this